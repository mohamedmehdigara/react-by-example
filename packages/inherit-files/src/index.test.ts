import { inheritFiles, Project } from '.'

const projects: Record<string, Project> = {
  noExtends: {
    'foo.txt': 'foo',
  },
  oneExtends: {
    '.extends': 'noExtends',
    'bar.txt': 'bar',
  },
  recursiveExtends: {
    '.extends': 'oneExtends',
    'baz.txt': 'baz',
  },
}

describe('inheritFiles', () => {
  const resolveProject = (id: string) => Promise.resolve(projects[id])

  describe('no extends', () => {
    it('should render project', async () => {
      expect(
        await inheritFiles('noExtends', { resolveProject }),
      ).toMatchSnapshot()
    })
  })

  describe('one extends', () => {
    it('should render project', async () => {
      expect(
        await inheritFiles('oneExtends', { resolveProject }),
      ).toMatchSnapshot()
    })
  })

  describe('recursive extends', () => {
    it('should render project', async () => {
      expect(
        await inheritFiles('recursiveExtends', { resolveProject }),
      ).toMatchSnapshot()
    })
  })
})
