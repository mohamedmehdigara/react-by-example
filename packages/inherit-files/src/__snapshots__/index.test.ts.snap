// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inheritFiles no extends should render project 1`] = `
Object {
  "foo.txt": "foo",
}
`;

exports[`inheritFiles one extends should render project 1`] = `
Object {
  "bar.txt": "bar",
  "foo.txt": "foo",
}
`;

exports[`inheritFiles recursive extends should render project 1`] = `
Object {
  "bar.txt": "bar",
  "baz.txt": "baz",
  "foo.txt": "foo",
}
`;
